src/openssl_wrap.cpp

```diff
@@ -1386,6 +1393,49 @@ SSL_HMAC_CTX_Ptr _HMAC_CTX_new()
     return createManagedOpenSSLObject<SSL_HMAC_CTX_Ptr>();
 }
 
+SSL_CMAC_CTX_Ptr _CMAC_CTX_new(void)
+{
+    return createManagedOpenSSLObject<SSL_CMAC_CTX_Ptr>();
+}
+
+void _CMAC_Init(CMAC_CTX *ctx, const std::vector<uint8_t> &key, const EVP_CIPHER *cipher, ENGINE *impl)
+{
+    OpensslCallIsOne::callChecked(
+            lib::OpenSSLLib::SSL_CMAC_Init, ctx,
+            key.data(), key.size(),
+            cipher,
+            impl);
+}
+
+void _CMAC_Update(CMAC_CTX *ctx, const std::vector<uint8_t> &data)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_CMAC_Update, ctx, data.data(), data.size());
+}
+
+std::vector<uint8_t> _CMAC_Final(CMAC_CTX *ctx)
+{
+    std::vector<uint8_t> cmac(EVP_MAX_BLOCK_LENGTH);
+    size_t length = 0;
+    OpensslCallIsOne::callChecked(
+            lib::OpenSSLLib::SSL_CMAC_Final, ctx,
+            cmac.data(), &length);
+    assert(length <= cmac.size());
```

Comment:
Can we statically guarantee that this will never happen? Otherwise it may be better to call `CMAC_Final` twice, once with a `nullptr` as buffer to obtain the required length, and then with a buffer of the required size.

Reply:
There is even an API available (`EVP_CIPHER_CTX_block_size()`) to get the block size of a specific cipher.
But yes, MAC size is specified as the block size and `EVP_MAX_BLOCK_LENGTH` represents the biggest block size of any cipher supported by OpenSSL. Looking into the OpenSSL source of `CMAC_Final()` also proves that this should work.

Do you prefer to use `EVP_CIPHER_CTX_block_size()`?

Reply:
No, this is fine as-is then.